src/error.rs

```diff
@@ -298,15 +358,36 @@ impl StdError for ExternError {
 impl ExternError {
     pub fn root_cause(&self) -> &(dyn StdError + 'static) {
         match self {
-            ExternError::Lua(err) => err,
+            ExternError::Lua { error, .. } => error,
             ExternError::Runtime(err) => err.root_cause(),
         }
     }
+
+    pub fn pretty_print(
+        &self,
+        writer: &mut impl fmt::Write,
+        source_map: Option<&SourceMap>,
+    ) -> Result<(), fmt::Error> {
```

Comment:
This may be easier to use if it returns `impl fmt::Display` rather than taking a Write implementation; for an example see https://github.com/kyren/piccolo/blob/ce709eb1dae5c543cbc78e7e12bb80249d88c55f/src/value.rs#L55-L59
(Actually, `pretty_print_error_with_backtrace` could itself return `impl fmt::Display` to reduce code duplication from monomorphization.)